### PR TITLE
Some code cleaning

### PR DESCRIPTION
**Pull request by bramtayl**

### CODE REVIEW

**Comment by tshort** [comment]
I'm generally good with these cleanups once tests pass.

**Comment by bramtayl** [comment]
Had to abandon MacroTools. I'll add more tests

**Comment by bramtayl** [comment]
Passed!

**Comment by tshort** [comment]
Great. I'll take a closer look tonight. It'll need squashed.

**Comment by ararslan** [comment]
The commits could be squashed on the merge using that fancy GitHub feature though, yeah?

File: src/DataFramesMeta.jl [hunk 1]
```diff
@@ -17,42 +17,45 @@ include("byrow.jl")
...
+onearg(e, f) = e.head == :call && length(e.args) == 2 && e.args[1] == f
+mapexpr(f, e) = Expr(e.head, map(f, e.args)...)
+
+replace_syms!(x, membernames) = x
+replace_syms!(e::Expr, membernames) =
+    onearg(e, :^)    ? e.args[2]                                       :
+    onearg(e, :_I_)  ? addkey!(membernames, :($(e.args[2])))           :
+    e.head == :quote ? addkey!(membernames, Meta.quot(e.args[1]) )     :
+    e.head == :.     ? replace_dotted!(e, membernames)                 :
+                       mapexpr(x -> replace_syms!(x, membernames), e)
```
Comment (ararslan) [inline src/DataFramesMeta.jl]: IMO this chain of ternaries would be more readable broken into `if`s, like
```julia
if onearg(e, :^)
    e.args[2]
elseif onearg(e, :_I_)
   ...
end
```

File: src/DataFramesMeta.jl [hunk 2]
```diff
@@ -17,42 +17,45 @@ include("byrow.jl")
...
+
+protect_replace_syms!(e, membernames) = e
+protect_replace_syms!(e::Expr, membernames) =
+  e.head == :quote ? e : replace_syms!(e, membernames)
```
Comment (ararslan) [inline src/DataFramesMeta.jl]: Four space indent please

File: src/DataFramesMeta.jl [hunk 3]
```diff
@@ -17,42 +17,45 @@ include("byrow.jl")
...
+
+function replace_dotted!(e, membernames)
+  x_new = replace_syms!(e.args[1], membernames)
+  y_new = protect_replace_syms!(e.args[2], membernames)
+  :($x_new.($y_new))
```
Comment (ararslan) [inline src/DataFramesMeta.jl]: Here too, should be four spaces

File: test/dataframes.jl [hunk 4]
```diff
@@ -27,12 +29,19 @@ idx2 = :B
 @test  @with(df, df[:A .> 1, ^([:B, :A])]) == df[df[:A] .> 1, [:B, :A]]
 @test  @with(df, DataFrame(a = :A * 2, b = :A + :B)) == DataFrame(a = df[:A] * 2, b = df[:A] + df[:B])
 
+where(df, 1) == df[1, :]
```
Comment (tshort) [inline test/dataframes.jl]: Needs an `@test` in front.

**Comment by tshort** [comment]
@ararslan, good point on squashing using GH's features. We'll do that. 

@bramtayl, looks good to go once my couple of bits and those by Alex are addressed.

**Comment by ararslan** [comment]
Alright, looks good to me now. Thanks for contributing, @bramtayl.

**Comment by nalimilan** [comment]
I wanted to merge, but the parent repo has been removed. Reopened as https://github.com/JuliaStats/DataFramesMeta.jl/pull/71.

**Pull request closed without merging**